app/logic/events.py

```diff
@@ -119,11 +112,19 @@ def getStudentLedProgram(term):
 
 def getTrainingProgram(term):
 
-    trainingEvents = (Event.select(Event, Program.id.alias("program_id"))
-                           .join(ProgramEvent)
-                           .join(Program)
+    allTrainingsEvent = ((ProgramEvent.select(ProgramEvent.event, fn.COUNT(1).alias('num_programs'))
+                                        .join(Event)
+                                        .group_by(ProgramEvent.event)
+                                        .order_by(fn.COUNT(1).desc()))
+                                        .where(Event.term == term).get())
+    trainingEvents = ((Event.select(Event)
+			               .order_by((Event.id == allTrainingsEvent.event.id).desc(), Event.startDate.desc())
                            .where(Event.isTraining,
-                                  Event.term == term))
+                                  Event.term == term)))
+
+    for t in trainingEvents:
+        print(t)
```

Comment:
debug output

Reply:
I think this disappeared when I merged development in